src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/NPathComplexityCheck.java

```diff
@@ -155,56 +196,249 @@ public void leaveToken(DetailAST ast) {
+        final BigInteger expressionValue = valuePair.getExpressionValue();
+        currentRangeValue = currentRangeValue
+                .subtract(BigInteger.ONE).add(basicRangeValue).add(expressionValue);
     }
 
-    /** Leaves else, catch or case. */
+    /**
+     * Process the end of a method definition.
+     * @param ast the token type representing the method definition
+     */
+    private void leaveMethodDef(DetailAST ast) {
+        final BigInteger bigIntegerMax = BigInteger.valueOf(max);
+        if (currentRangeValue.compareTo(bigIntegerMax) > 0) {
+            log(ast, MSG_KEY, currentRangeValue, bigIntegerMax);
+        }
+        popValue();
+        currentRangeValue = INITIAL_VALUE;
+    }
+
+    /** Leaves catch. */
     private void leaveAddingConditional() {
-        currentValue = currentValue.subtract(BigInteger.ONE).add(popValue());
+        currentRangeValue = currentRangeValue.subtract(BigInteger.ONE)
+                .add(popValue().getRangeValue());
     }
 
-    /** Visits while, do, for, if, try, ? (in ?::) or switch. */
-    private void visitMultiplyingConditional() {
-        pushValue();
+    /**
+     * Pushes the current range value on the range value stack. Pushes this token expression value
+     * on the expression value stack.
+     * @param expressionValue value of expression calculated for current token.
+     */
+    private void pushValue(Integer expressionValue) {
+        rangeValues.push(currentRangeValue);
+        expressionValues.push(expressionValue);
+        currentRangeValue = INITIAL_VALUE;
     }
 
-    /** Leaves while, do, for, if, try, ? (in ?::) or switch. */
+    /**
+     * Pops values from both stack of expression values and stack of range values.
+     * @return pair of head values from both of the stacks.
+     */
+    private Values popValue() {
+        final int expressionValue = expressionValues.pop();
+        return new Values(rangeValues.pop(), BigInteger.valueOf(expressionValue));
+    }
+
+    /** Leaves try. */
     private void leaveMultiplyingConditional() {
-        currentValue = currentValue.add(BigInteger.ONE).multiply(popValue());
+        currentRangeValue = currentRangeValue.add(BigInteger.ONE)
+                .multiply(popValue().getRangeValue());
     }
 
-    /** Push the current value on the stack. */
-    private void pushValue() {
-        valueStack.push(currentValue);
-        currentValue = INITIAL_VALUE;
+    /**
+     * Calculates number of logical operators, including inline ternary operatior, for a token.
+     * @param ast inspected token.
+     * @return number of logical operators.
+     */
+    private static int countLogicalOperators(DetailAST ast) {
+        int number = 0;
+        for (DetailAST child = ast.getFirstChild(); child != null;
+                child = child.getNextSibling()) {
+            final int type = child.getType();
+            if (type == TokenTypes.LOR || type == TokenTypes.LAND) {
```

Comment:
All information that was missed and helped you to write correct code should be placed  to javadoc, next contributor might trap in the same problem.
Link to jdk should be in javadoc